storage/client_raft_test.go

```diff
@@ -863,5 +864,6 @@ func TestRaftAfterRemoveRange(t *testing.T) {
 		t.Fatal(err)
 	}
 	// Execute another replica change to ensure that MultiRaft has processed the heartbeat just sent.
+	// TODO(tschottdorf): without this line, test passes. Gotta investigate.
```

Comment:
OK, so it looks like a timing issue that was exposed by the fact that `EndTransaction` now takes longer. It has to do with coalesced heartbeats. After range 2 is removed, node 2 believes that it has no ranges in common with node 1, so it doesn't respond to heartbeats. We have a special case to send heartbeats when the receiving node doesn't know anything about the sending node, but nothing for when the recipient doesn't think there are any ranges in common. I think we just need to remove the `cnt` check from `fanoutHeartbeat` and reply unconditionally.